### ISSUE TITLE
Multischema configuration does not support environment variables

### ISSUE THREAD

**Issue by aaron-henselin** [prisma/prisma#15089]
When attempting to use an **environment variable** to configure the schemas array in the datasource, OR the @@schema for models, the following error occurs:

```
Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
Error: Schema validation error - Error (query-engine-node-api library)
Error code: P1012
error: Expected a string value, but received functional value `env("DATABASE_SCHEMA")`.
  -->  schema.prisma:10
   | 
 9 |   url      = env("DATABASE_URL")
10 |   schemas  = [env("DATABASE_SCHEMA"), "public"]
   | 

Validation Error Count: 1
[Context: getConfig]

```

**Comment by janpio** [comment]
That is currently intentional and not implemented. What is your use case that would make this useful?

**Comment by aaron-henselin** [comment]
In this case, Heroku Connect, which uses the database as a formal pass-through into Salesforce. 
A Heroku Connect connected database can have "copies" of the same tables which direct the bidirectional replication, each in its own schema.  So instead of a traditional "app1" / "app2" schema, it's a "dev" / "production" schema. 
Is there a workaround or better way you can think of to accommodate this?

**Comment by janpio** [comment]
Interesting. So you would use the `env` in the `schemas` list to essentially be able to switch between your dev and production data?

**Comment by aaron-henselin** [comment]
Exactly. In this use case, schema is really treated as a part of the environment/connectionstring rather than the application.

**Comment by jwwisgerhof** [comment]
We have the same issue. Our primary shema's name has a postfix that denotes the environment. Even though our production schema is in a separate DB, our development schemas (we run ~4 test instances) run in the same DB server as cost saving.

Most critically, we receive external (integration) data from other organisational units as a direct DB injection - they determine the database name.

**Comment by janpio** [comment]
@aaron-henselin From you responses I assume you used `multiSchema` with a Heroku Connect database already. Can you maybe tell us a bit about your experience in this Heroku Connect specific issue https://github.com/prisma/prisma/issues/7801? Thanks!

**Comment by ZeeshanZubair2** [comment]
I have the same kind of issue. I am implementing schema level multitenancy. Each tenant has its own schema. On the basis of request I would connect to a specific schema.
`import { PrismaClient } from '@prisma/client'
const client = new PrismaClient({ datasources: { db: { url: 'postgresql://postgres:postgres@localhost:5432/db?schema=${request.tenantId}' }} })`
I want to use URL schema in schemas array. Something like this
`datasource database {
  provider = "postgresql"
  url      = env("POSTGRES_DATABASE_URL")
  schemas  = ["common",@defaultSchema]
}

generator client {
  provider = "prisma-client-js"
  output   = "../node_modules/@notiz/prisma"
  previewFeatures = ["multiSchema"]
}

model test {
  userId     String   @map("user_id") @database.Uuid
  interestId String   @map("interest_id")
  @@id([userId, interestId])
  @@schema("common")
}

model user {
  id String
  name String
   @@schema(@defaultSchema)
}`

**Comment by salim-runsafe** [comment]
We are running into this issue while migrating from a different ORM. Our use case is that for CI pipeline tests we deploy nearly-identical ephemeral environments - one per development branch - separated by DB schemas within the same DB. Most tables are separated out into this different schemas but a few tables that change infrequently and have a large amount of data are in a schema common to all deployments.

We want to be able to provide two variable schemas within the prisma datasource and provide variable `@@schema` definitions per model.

@janpio before I spend time looking at the code, would a PR be welcome for this?

**Comment by FGoessler** [comment]
We have some ideas on how to enable this but they will NOT be implemented before moving the multi schema preview feature to general availability in the next weeks.
